mesos-rxjava-example/mesos-rxjava-example-framework/src/main/java/org/apache/mesos/rx/java/example/framework/sleepy/Main.java

```diff
@@ -170,11 +117,55 @@ private static void _main(final String[] args) throws InterruptedException {
         final Subscription sink = client.sink(calls);
         Thread.sleep(Integer.MAX_VALUE); //TODO: Figure out something better than this
         sink.unsubscribe();
-        setFrameworkIdSubscription.unsubscribe();
         errorLoggerSubscription.unsubscribe();
     }
 
     @NotNull
+    private static SinkOperation<Call> handleOffer(@NotNull final Tuple2<Offer, State> t) {
+        final Offer offer = t._1;
+        final State state = t._2;
+        final int offerCount = state.getOfferCounter().incrementAndGet();
+
+        final FrameworkID frameworkId = state.getFwId();
+        final AgentID agentId = offer.getAgentId();
+        final List<OfferID> ids = newArrayList(offer.getId());
+
+        final Map<String, List<Resource>> resources = offer.getResourcesList()
+            .stream()
+            .collect(groupingBy(Resource::getName));
+        final List<Resource> cpuList = resources.get("cpus");
+        final List<Resource> memList = resources.get("mem");
+        if (cpuList != null && !cpuList.isEmpty() && memList != null && !memList.isEmpty()) {
+            final Resource cpus = cpuList.iterator().next();
+            final Resource mem = memList.iterator().next();
+            final List<TaskInfo> tasks = newArrayList();
+
+            double availableCpu = cpus.getScalar().getValue();
+            double availableMem = mem.getScalar().getValue();
+            final double cpusPerTask = state.getCpusPerTask();
+            final double memMbPerTask = state.getMemMbPerTask();
+            while (availableCpu >= cpusPerTask && availableMem >= memMbPerTask) { // .foldLeft((newArrayList(), cpu, mem))
```

Comment:
Not dead code, a comment for me to easily remember what that `while` look was doing. (I've gotten too used to functional stuff and find manual loops harder to read now). I'll remove it since it looks like dead code.